orangecontrib/text/widgets/owdocumentembedding.py

```diff
@@ -0,0 +1,236 @@
+    _auto_apply = Setting(default = True)
+
+    class Inputs:
+        corpus = Input('Corpus', Corpus)
+
+    class Outputs:
+        new_corpus = Output('Corpus', Corpus)
+
+    class Error(OWWidget.Error):
+        no_connection = Msg("""No internet connection.
+                            Please establish a connection or 
+                            use another vectorizer.""")
+        unexpected_error = Msg('Embedding error: {}')
+
+    class Warning(OWWidget.Warning):
+        unsuccessful_embeddings = Msg('Some embeddings were unsuccessful.')
+
+    
+    language = Setting(default = 'English')
+    aggregator = Setting(default = 'mean')
+
+    def __init__(self):
+        OWWidget.__init__(self)
+        ConcurrentWidgetMixin.__init__(self)
+
+        self.languages = list(LANGS_TO_ISO.keys())
+        self.aggregators = AGGREGATORS
+        self.corpus = None
+        self.new_corpus = None
+        self._setup_layout()
+
+    def _setup_layout(self):
+        self.controlArea.setMinimumWidth(self.controlArea.sizeHint().width())
+
+        widget_box = widgetBox(self.controlArea, 'Settings')
+
+        self.language_cb = comboBox(widget = widget_box,
+                                    master = self,
+                                    value = 'language',
+                                    label = 'Language: ',
+                                    orientation = Qt.Horizontal,
+                                    items = self.languages,
+                                    callback = self._option_changed
+        )
+
+        self.aggregator_cb = comboBox(widget = widget_box,
+                                      master = self,
+                                      value = 'aggregator',
+                                      label = 'Aggregator: ',
+                                      orientation = Qt.Horizontal,
+                                      items = self.aggregators,
+                                      callback = self._option_changed 
+        )
+
+        self.auto_commit_widget = auto_commit(widget = self.controlArea,
+                                              master = self,
+                                              value = '_auto_apply',
+                                              label = 'Apply',
+                                              commit = self.commit)
+
+        self.cancel_button = QPushButton(
+                                    'Cancel',
+                                    icon = self.style()
+                                    .standardIcon(QStyle.SP_DialogCancelButton))
+
+        self.cancel_button.clicked.connect(self.cancel)
+
+        hbox = hBox(self.controlArea)
+        hbox.layout().addWidget(self.cancel_button)
+        self.cancel_button.setDisabled(True)
+
+    def set_input_corpus_summary(self, corpus):
+        if(corpus is None):
+            self.info.set_input_summary(self.info.NoInput)
+        else:
+            self.info.set_input_summary("{} documents."
```

Comment:
This string is too long for a status bar. The prefered solution is to give the function `set_input_summary` two parameters: first is just length of corpus and the second is a string with details:
```
self.info.set_input_summary(len(corpus), "{} documents.".format(len(corpus)))
```
The string is the second argument is visible when the user clicks on input summary.